python/ray/_private/node.py

```diff
@@ -83,6 +83,9 @@ def __init__(
         )
         self.all_processes: dict = {}
         self.removal_lock = threading.Lock()
+        self.cluster_id = (
+            ray_params.cluster_id if hasattr(ray_params, "cluster_id") else None
```

Comment:
Originally it was because there was no enforcement that the attr was even declared, and when it's not declared, it throws exception instead of returning None (i.e. if `ray_params.cluster_id = cluster_id` is only called sometimes). 

But RayParams is fixed now per your other comment.